cranelift/codegen/src/egraph/cost.rs

```diff
@@ -43,12 +86,49 @@ impl Cost {
         Cost(0)
     }
 
+    fn new(opcode_cost: u32, depth: u8) -> Cost {
+        debug_assert!(
+            opcode_cost <= Self::MAX_OP_COST,
```

Comment:
Should this be strictly less-than? If the value is equal to `MAX_OP_COST`, and `depth` is `255`, we'll allow constructing the special `infinity` value.

Reply:
Reworked things a bit to make constructing infinity by accident impossible.